core/store/models/cbor_test.go

```diff
@@ -120,11 +128,105 @@ func Test_autoAddMapDelimiters(t *testing.T) {
 	}
 }
 
-func jsonMustUnmarshal(in string) JSON {
+func jsonMustUnmarshal(t *testing.T, in string) JSON {
 	var j JSON
 	err := json.Unmarshal([]byte(in), &j)
-	if err != nil {
-		log.Panicf("Failed to unmarshal '%s'", in)
-	}
+	require.NoError(t, err)
 	return j
 }
+
+func TestCoerceInterfaceMapToStringMap(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name      string
+		input     interface{}
+		want      interface{}
+		wantError bool
+	}{
+		{"empty map", map[interface{}]interface{}{}, map[string]interface{}{}, false},
+		{"simple map", map[interface{}]interface{}{"key": "value"}, map[string]interface{}{"key": "value"}, false},
+		{"int map", map[int]interface{}{1: "value"}, map[int]interface{}{1: "value"}, false},
+		{"error map", map[interface{}]interface{}{1: "value"}, map[int]interface{}{}, true},
+		{
+			"nested string map map",
+			map[string]interface{}{"key": map[interface{}]interface{}{"nk": "nv"}},
+			map[string]interface{}{"key": map[string]interface{}{"nk": "nv"}},
+			false,
+		},
+		{
+			"nested map map",
+			map[interface{}]interface{}{"key": map[interface{}]interface{}{"nk": "nv"}},
+			map[string]interface{}{"key": map[string]interface{}{"nk": "nv"}},
+			false,
+		},
+		{
+			"nested map array",
+			map[interface{}]interface{}{"key": []interface{}{1, "value"}},
+			map[string]interface{}{"key": []interface{}{1, "value"}},
+			false,
+		},
+		{"empty array", []interface{}{}, []interface{}{}, false},
+		{"simple array", []interface{}{1, "value"}, []interface{}{1, "value"}, false},
+		{
+			"error array",
+			[]interface{}{map[interface{}]interface{}{1: "value"}},
+			[]interface{}{},
+			true,
+		},
+		{
+			"nested array map",
+			[]interface{}{map[interface{}]interface{}{"key": map[interface{}]interface{}{"nk": "nv"}}},
+			[]interface{}{map[string]interface{}{"key": map[string]interface{}{"nk": "nv"}}},
+			false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			decoded, err := CoerceInterfaceMapToStringMap(test.input)
+			if test.wantError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.True(t, reflect.DeepEqual(test.want, decoded))
+			}
+		})
+	}
+}
+
+func TestJSON_CBOR(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		in   JSON
+	}{
+		{"empty object", JSON{}},
+		{"array", jsonMustUnmarshal(t, `[1,2,3,4]`)},
+		{
+			"hello world",
```

Comment:
is there a better name for this?

Reply:
Hmm, this is an old test, so I'll take a crack at it: how about "simple object" ?

Reply:
haha sounds great :+1: only realized later that a lot of this code was just moved around